src/components/pages/CheckoutAuthPage/CheckoutAuthPage.jsx

```diff
@@ -69,8 +71,25 @@ class CheckoutAuthPage extends React.Component {
     }
     submitEmail(event) {
         event.preventDefault();
-        // TODO: submit the email to form, and validate it's actually an email (*@*.*)
-        console.log(this.props.history);
+        login().then(() => {
+            fetch(Config.cortexApi.path + '/emails/' + Config.cortexApi.scope + '/form', {
+                method: 'post',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                },
+                body: JSON.stringify({ 'email': this.state.email })
+            }).then(res => {
+                if (res.status === 400) {
+                    this.setState({badEmail: true});
+                } else if (res.status === 201) {
+                    this.setState({badEmail: false});
+                    this.props.history.push('/checkout');
```

Comment:
Do you care here that setState is run asynchronously?

Reply:
no, you make a good point. making this change, and in a couple other spots.